src/GraphRequestUtil.ts

```diff
@@ -58,3 +58,99 @@ export const serializeContent = (content: any): any => {
 	}
 	return content;
 };
+
+export class PathUtils {
+	private static hostRe = /([a-z]+.microsoft.[a-z]+)/g;
+	private static versionRe = /(beta|v1.0)/g;
```

Comment:
Q: how do we guarantee that false positives won't appear in other parts of the url? For example:
`https://graph.microsoft.com/v1.0/me/messages?$filter=startswith(subject, 'anothermatchingurl.microsoft.com')`
or
`https://graph.microsoft.com/v1.0/me/messages?$filter=startswith(subject, 'beta')`

Reply:
This will match multiple groups for example `graph.microsoft.com` and `anothermatchingurl.microsoft.com` but the getHostMethod will return the first match which is `graph.microsoft.com`